accepted/build/dotnet-patch-check.md

```diff
@@ -0,0 +1,27 @@
+# Getting latest .NET Core Runtime patch information
+
+## Goal
+
+When patches are released for the .NET Core runtime, developers should not need to install an entirely new .NET Core SDK.  They should be able to install only the updated runtime as needed, and the existing SDK should have a way of discovering that new patches should be used for self-contained applications.
+
+Eventually we hope to be able to stop producing new releases of the .NET Core SDK whenever there is a patch to the runtime.  This proposal is a step in that direction, however a lot more will be needed to enable that.
+
+## Proposal overview
+
+- Add a `dotnet` command which will check for the latest patches to the .NET Core runtime.  Naming is TBD, straw man for this proposal is `dotnet patch-check`.  The command will:
+    - Update the latest patch versions to use for self-contained applications
+    - Notify the user if the installed shared frameworks are not up-to-date, with a link to download the latest patch
+- When building a self-contained app, generate a warning if `dotnet patch-check` has not been run within a specified period of time (7 days?), suggesting that the user run `dotnet patch-check` to make sure the self-contained app uses the latest patch
```

Comment:
Why is this not just part of `dotnet restore` for self-contained app? The package for self-contained runtime is no different from other NuGet packages that may have servicing security fixes as well.

Reply:
We've looked at that and it ends up being expensive to ask for the latest version of a NuGet package on every build.  As I understand it, when a specific package version is requested, then there doesn't even need to be a web request if that package is already in the local packages folder.  However, if you want to ask NuGet for the latest version, then it has to request that from all the configured feeds, and wait for a response from all of them.

So this proposal is an attempt to sidestep those issues by having an explicit command to check for updates.  It should be an improvement over today's status quo, which is that you need to install an entirely new SDK to get the latest runtime patches by default.

Reply:
I understand that it is expensive to ask for the latest version of a NuGet package on every build.

My point is that there should be just a single gesture required to ensure that all my app dependencies are up to date. I do not think it makes sense to have one gesture for the runtime package itself and a different gesture for the NuGet packages that my app depends on.